nucypher/utilities/prometheus/metrics.py

```diff
@@ -0,0 +1,313 @@
+        },
+        argument_filters={'staker': ursula.checksum_address},
+        contract_agent=staking_agent))
+
+    # Minted
+    collectors.append(EventMetricsCollector(
+        event_name='Minted',
+        event_args_config={
+            "value": (Gauge, f'{metrics_prefix}_mined_value', 'Mined value'),
+            "period": (Gauge, f'{metrics_prefix}_mined_period', 'Mined period'),
+            "block_number": (Gauge, f'{metrics_prefix}_mined_block_number', 'Mined block number')
+        },
+        argument_filters={'staker': ursula.checksum_address},
+        contract_agent=staking_agent))
+
+    # Slashed
+    collectors.append(EventMetricsCollector(
+        event_name='Slashed',
+        event_args_config={
+            "penalty": (Gauge, f'{metrics_prefix}_last_slashed_penalty', 'Penalty for slashing'),
+            "block_number": (Gauge, f'{metrics_prefix}_last_slashed_penalty_block_number', 'Slashed penalty block number')
+        },
+        argument_filters={'staker': ursula.checksum_address},
+        contract_agent=staking_agent))
+
+    # RestakeSet
+    collectors.append(ReStakeEventMetricsCollector(
+        event_args_config={
+            "reStake": (Gauge, f'{metrics_prefix}_restaking', 'Restake set')
+        },
+        argument_filters={'staker': ursula.checksum_address},
+        staker_address=ursula.checksum_address,
+        contract_agent=staking_agent))
+
+    # WindDownSet
+    collectors.append(WindDownEventMetricsCollector(
+        event_args_config={
+            "windDown": (Gauge, f'{metrics_prefix}_wind_down', 'is windDown')
+        },
+        argument_filters={'staker': ursula.checksum_address},
+        staker_address=ursula.checksum_address,
+        contract_agent=staking_agent))
+
+    # WorkerBonded
+    collectors.append(WorkerBondedEventMetricsCollector(
+        event_args_config={
+            "startPeriod": (Gauge, f'{metrics_prefix}_worker_set_start_period', 'New worker was set'),
+            "block_number": (Gauge, f'{metrics_prefix}_worker_set_block_number', 'WorkerSet block number')
+        },
+        argument_filters={'staker': ursula.checksum_address},
+        staker_address=ursula.checksum_address,
+        worker_address=ursula.worker_address,
+        contract_agent=staking_agent))
+
+    return collectors
+
+
+def create_worklock_events_metric_collectors(ursula: Ursula, metrics_prefix: str) -> List[MetricsCollector]:
+    collectors: List[MetricsCollector] = []
+    worklock_agent = ContractAgency.get_agent(WorkLockAgent, registry=ursula.registry)
+
+    # Deposited
+    collectors.append(EventMetricsCollector(
+        event_name='Deposited',
+        event_args_config={
+            "value": (Gauge, f'{metrics_prefix}_worklock_deposited_value', 'Deposited value')
+        },
+        argument_filters={"sender": ursula.checksum_address},
+        contract_agent=worklock_agent))
+
+    # Claimed
+    collectors.append(EventMetricsCollector(
```

Comment:
this is looking pretty tidy but I am still wondering if it is possible to move this data generation to a (static?)method on each collector class.